client/Elements/Components/CreateTopic/StepMessageRetention.patternfly.tsx

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+import React, { useState } from 'react';
+import '@patternfly/react-core/dist/styles/base.css';
+import {
+  Flex,
+  FlexItem,
+  Form,
+  FormGroup,
+  Radio,
+  Select,
+  SelectOption,
+  SelectVariant,
+  TextContent,
+  Text,
+  TextVariants,
+  Touchspin,
+} from '@patternfly/react-core';
+import './CreateTopicWizard.patternfly.css';
+
+interface IStepMessageRetention {
+  setMsgRetentionValue: (value: number) => void;
+}
+
+export const StepMessageRetention: React.FC<IStepMessageRetention> = ({
+  setMsgRetentionValue,
+}) => {
+  const [radioBtnDay, setRadioBtnDay] = useState(false);
+  const [radioBtnWeek, setRadioBtnWeek] = useState(false);
+  const [radioBtnMonth, setRadioBtnMonth] = useState(false);
+  const [radioBtnCustomTime, setRadioBtnCustomTime] = useState(false);
+  const [msgTouchspinValue, setMsgTouchspinValue] = useState(7);
+  const [isMsgSelectOpen, setIsMsgSelectOpen] = useState(false);
+  const [selected, setSelected] = useState(false);
+  let retentionFactor = 1;
+  const handleChangeStep3 = (checked, event) => {
+    setRadioBtnDay(false);
+    setRadioBtnWeek(false);
+    setRadioBtnMonth(false);
+    setRadioBtnCustomTime(false);
+
+    const target = event.target;
+    const name = target.name;
+
+    if (name === 'radio1') {
+      setRadioBtnDay(true);
+      setMsgRetentionValue(1); //A day
+    } else if (name === 'radio2') {
+      setRadioBtnWeek(true);
+      setMsgRetentionValue(7); //A week
+    } else if (name === 'radio3') {
+      setRadioBtnMonth(true);
+      setMsgRetentionValue(30); //A month
+    } else if (name === 'radio4') {
+      setRadioBtnCustomTime(true);
+      setMsgRetentionValue(retentionFactor * msgTouchspinValue);
+    }
+  };
+
+  const onMsgToggle = (isMsgSelectOpen) => {
+    setIsMsgSelectOpen(isMsgSelectOpen);
+  };
+
+  const onMsgSelect = (event, selection) => {
+    if (selection === 'days') {
+      retentionFactor = 1;
+    } else if (selection === 'weeks') {
+      retentionFactor = 7;
+    } else if (selection === 'months') {
+      retentionFactor = 30;
+    }
```

Comment:
Simply set the currentPeriod, use a React.useEffect to handle changes to either period or value to then call `setMsgRetentionValue`